packages/app/src/components/Randomizer/Randomizer.js

```diff
@@ -1,58 +1,111 @@
-import React, { useState } from 'react'
+import React, { useState, useEffect } from 'react'
 import { Button } from 'react-bootstrap'
 import { ExpandMore, ExpandLess } from '@material-ui/icons'
+import PropTypes from 'prop-types'
 import Filter from '../Filter/Filter/Filter'
 import RestaurantEntry from '../RestaurantEntry/RestaurantEntry'
 import soundService from '../../services/sound'
 import restaurantService from '../../services/restaurant'
 import './Randomizer.css'
 
 
-const Randomizer = () => {
-  const maxNumberOfRotations = 28
-  const minTimeBetweenRotations = 15 // in milliseconds
-  const [restaurant, setRestaurant] = useState({ name: 'Press the button' })
-  const [disableButton, setDisableButton] = useState(false)
+const Randomizer = ({ maxNumberOfRotations = 35, minTimeBetweenRotations = 25, maxTimeBetweenRotations = 1000, resultWaitTime = 1250 }) => {
+  const [restaurant, setRestaurant] = useState()
+  const [isRolling, setRolling] = useState(false)
   const [filterType, setFilterType] = useState('some')
   const [filterCategories, setFilterCategories] = useState([])
   const [distance, setDistance] = useState('')
   const [showFilter, setShowFilter] = useState(false)
 
-  const sleep = async (duration) => {
-    return new Promise(r => setTimeout(r, duration))
+  const [timeoutHandle, setTimeoutHandle] = useState()
+  const [nope, setNope] = useState()
```

Comment:
Not very self-explanatory...

Reply:
`nope` is the hidden *"Easter egg"* feature which re-starts the roll, so reasoning for the name was to make the feature a bit more "hidden" by obfuscating the used state.

For code readability, this is just plain stupid. I'll change these for something more descriptive.